jscomp/super_errors/super_typetexp.ml

```diff
@@ -159,7 +159,11 @@ let setup () =
   Location.register_error_of_exn
     (function
       | Typetexp.Error (loc, env, err) ->
-        Some (Location.error_of_printer loc (report_error env) err)
+        Some (Location.error_of_printer loc (fun ppf ->
```

Comment:
@bobzhang so, I need to detect whether a file's in Reason before printing the error in Reason syntax. But I can't find a higher-level common point to do this instead of duplicating the outcome printer initialization logic everywhere. Putting it in `Super_location.super_error_reporter` was too late in terms of printing, apparently.

Some other files like `typedecl.ml` would also need to be wrapped this way. If this is a good way to go then I can add them (or lazily add them in a later diff when I do need to change typedecl)